src/shared/modules/dbMeta/dbMetaDuck.ts

```diff
@@ -658,9 +663,15 @@ export const serverConfigEpic = (some$: any, store: any) =>
           store.dispatch(
             updateUserCapability(USER_CAPABILITIES.serverConfigReadable, true)
           )
+          const prevShouldRetainHistory = shouldRetainEditorHistory(
+            store.getState()
+          )
           store.dispatch(updateSettings(settings))
-          // settings must be updated in state for this check to work
-          if (!shouldRetainEditorHistory(store.getState())) {
+          const shouldRetainHistory = shouldRetainEditorHistory(
+            store.getState()
+          )
+          // if setting changed either prev/next was false so clear history
+          if (prevShouldRetainHistory !== shouldRetainHistory) {
```

Comment:
It's not clear to me if we should clear the history here? 
Move it in / out of local storage, yes.
Remove it from runtime state, why?

Reply:
This is a fail safe for when the connection is switched to a different server without the connection events.